### PR TITLE
Fixes for issue #74 and #75

### PR DESCRIPTION
**Pull request by lucdekens**
#74 Added class definition for ManagedObjectReference
     HostConfigManager now contains ManagedObjectReference entries

 #75 The Get-View function in the TestHelpers module VMware.VimAutomation.Core now supports all parametersets

Signed-off-by: Luc Dekens <dekens.luc@gmail.com>

### CODE REVIEW

File: Source/VMware.vSphereDSC/Tests/Unit/VMHostNtpSettings.Unit.Tests.ps1 [hunk 1]
```diff
@@ -622,7 +639,10 @@ Describe 'VMHostNtpSettings\Test' -Tag 'Test' {
             }
             $vmHostMock = {
                 return [VMware.Vim.VMHost] @{ ExtensionData = [VMware.Vim.HostExtensionData] @{ Config = [VMware.Vim.HostConfig] @{ DateTimeInfo `
-                     = [VMware.Vim.HostDateTimeConfig] @{ NtpConfig = [VMware.Vim.HostNtpConfig] @{} } } } }
+                                = [VMware.Vim.HostDateTimeConfig] @{ NtpConfig = [VMware.Vim.HostNtpConfig] @{} }
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/Tests/Unit/VMHostNtpSettings.Unit.Tests.ps1]: I think the brackets here are ok to stay on the same line as before.

File: Source/VMware.vSphereDSC/Tests/Unit/VMHostNtpSettings.Unit.Tests.ps1 [hunk 2]
```diff
@@ -638,8 +658,8 @@ Describe 'VMHostNtpSettings\Test' -Tag 'Test' {
 
             # Assert
             Assert-MockCalled -CommandName Connect-VIServer `
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/Tests/Unit/VMHostNtpSettings.Unit.Tests.ps1]: The parameters should stay each below the previous for consistency.

File: Source/VMware.vSphereDSC/Tests/Unit/VMHostNtpSettings.Unit.Tests.ps1 [hunk 3]
```diff
@@ -648,8 +668,8 @@ Describe 'VMHostNtpSettings\Test' -Tag 'Test' {
 
             # Assert
             Assert-MockCalled -CommandName Get-VMHost `
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/Tests/Unit/VMHostNtpSettings.Unit.Tests.ps1]: The same as the above comment.

File: Source/VMware.vSphereDSC/Tests/Unit/VMHostNtpSettings.Unit.Tests.ps1 [hunk 4]
```diff
@@ -661,7 +681,10 @@ Describe 'VMHostNtpSettings\Test' -Tag 'Test' {
             }
             $vmHostMock = {
                 return [VMware.Vim.VMHost] @{ ExtensionData = [VMware.Vim.HostExtensionData] @{ Config = [VMware.Vim.HostConfig] @{ DateTimeInfo `
-                     = [VMware.Vim.HostDateTimeConfig] @{ Id = 'HostDateTimeConfig'; NtpConfig = [VMware.Vim.HostNtpConfig] @{} } } } }
+                                = [VMware.Vim.HostDateTimeConfig] @{ Id = 'HostDateTimeConfig'; NtpConfig = [VMware.Vim.HostNtpConfig] @{} }
+                        }
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/Tests/Unit/VMHostNtpSettings.Unit.Tests.ps1]: The same comment for the brackets.

File: Source/VMware.vSphereDSC/Tests/Unit/VMHostNtpSettings.Unit.Tests.ps1 [hunk 5]
```diff
@@ -687,9 +710,14 @@ Describe 'VMHostNtpSettings\Test' -Tag 'Test' {
             }
             $vmHostMock = {
                 return [VMware.Vim.VMHost] @{ ExtensionData `
-                     = [VMware.Vim.HostExtensionData] @{ Config = [VMware.Vim.HostConfig] @{ DateTimeInfo = [VMware.Vim.HostDateTimeConfig] @{ Id = 'HostDateTimeConfig'; NtpConfig `
-                     = [VMware.Vim.HostNtpConfig] @{ Server = @('0.bg.pool.ntp.org') } } }; ConfigManager = [VMware.Vim.HostConfigManager] @{ DateTimeSystem `
-                     = [VMware.Vim.HostDateTimeSystem] @{ Id = 'HostDateTimeSystem' } } } }
+                        = [VMware.Vim.HostExtensionData] @{ Config = [VMware.Vim.HostConfig] @{ DateTimeInfo = [VMware.Vim.HostDateTimeConfig] @{ Id = 'HostDateTimeConfig'; NtpConfig `
+                                    = [VMware.Vim.HostNtpConfig] @{ Server = @('0.bg.pool.ntp.org') }
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/Tests/Unit/VMHostNtpSettings.Unit.Tests.ps1]: Here also the brackets could be on the same line and the same for the rest of the file.

File: Source/VMware.vSphereDSC/Tests/Unit/VMHostNtpSettings.Unit.Tests.ps1 [hunk 6]
```diff
@@ -866,26 +913,28 @@ Describe 'VMHostNtpSettings\Test' -Tag 'Test' {
...
-                     = [VMware.Vim.HostServiceInfo] @{ Service = @([VMware.Vim.HostService] @{ Key = 'ntpd'; Policy = 'automatic' }) } }; ConfigManager = [VMware.Vim.HostConfigManager] @{ ServiceSystem `
-                     = [VMware.Vim.HostServiceSystem] @{ Id = 'HostServiceSystem' } } } }
+                     = [VMware.Vim.HostServiceInfo] @{ Service = @([VMware.Vim.HostService] @{ Key = 'ntpd'; Policy = 'automatic' }) } }; ConfigManager `
+                            = [VMware.Vim.HostConfigManager] @{ ServiceSystem = [VMware.Vim.ManagedObjectReference] @{ Type = 'HostServiceSystem'; Value = 'ServiceSystem' } }
+                    }
+                }
             }
 
-            $script:resourceProperties.NtpServicePolicy = 'on'
+                $script:resourceProperties.NtpServicePolicy = 'on'
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/Tests/Unit/VMHostNtpSettings.Unit.Tests.ps1]: The spaces here look wrong,

File: Source/VMware.vSphereDSC/Tests/Unit/vCenterStatistics.Unit.Tests.ps1 [hunk 7]
```diff
@@ -62,18 +62,24 @@ Describe 'vCenterStatistics\Set' -Tag 'Set' {
     Context 'Invoking with default resource properties' {
         BeforeAll {
             $vCenter = [VMware.Vim.VCenter] @{ Name = '10.23.82.112'; User = 'user' }
+            $perfManagerMoRef = [VMware.Vim.ManagedObjectReference] @{ Type = 'PerformanceManager'; Value = 'PerfMgr' }
             $perfManager = [VMware.Vim.PerformanceManager] @{ Id = 'PerfManager Id' }
             $perfInterval = [VMware.Vim.PerfInterval] @{ Key = 1; Name = 'Month'; Enabled = $false; SamplingPeriod = 600; Length = 2629800; Level = 1 }
 
             # Arrange
             $vCenterMock = {
                 return [VMware.Vim.VCenter] @{ Name = '10.23.82.112'; User = 'user'; ExtensionData = `
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/Tests/Unit/vCenterStatistics.Unit.Tests.ps1]: The same for the brackets as the previous file.

File: Source/VMware.vSphereDSC/Tests/Unit/vCenterStatistics.Unit.Tests.ps1 [hunk 8]
```diff
@@ -94,8 +100,8 @@ Describe 'vCenterStatistics\Set' -Tag 'Set' {
 
             # Assert
             Assert-MockCalled -CommandName Connect-VIServer `
-                              -ParameterFilter { $Server -eq $script:resourceProperties.Server -and $Credential -eq $script:resourceProperties.Credential } `
-                              -ModuleName $script:moduleName -Exactly 1 -Scope It
+                -ParameterFilter { $Server -eq $script:resourceProperties.Server -and $Credential -eq $script:resourceProperties.Credential } `
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/Tests/Unit/vCenterStatistics.Unit.Tests.ps1]: And for the properties also as the previous file.

File: Source/VMware.vSphereDSC/Tests/Unit/TestHelpers/VMware.VimAutomation.Core/VMware.VimAutomation.Core.psm1 [hunk 9]
```diff
@@ -649,9 +671,25 @@ function Get-VMHost {
...
 
 function Get-View {
+    [CmdletBinding()]
     param(
         [PSObject] $Server,
-        [PSObject] $VIObject
+        [Parameter(ParameterSetName = 'GetViewByVIObject')]
+        [PSObject] $VIObject,
+        [Parameter(ParameterSetName = 'GetView')]
+        [VMware.Vim.ManagedObjectReference] $Id,
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/Tests/Unit/TestHelpers/VMware.VimAutomation.Core/VMware.VimAutomation.Core.psm1]: The type here should be [VMware.Vim.ManagedObjectReference[]].

**Comment by lucdekens** [comment]
The duplicate line with networkSystemMoRef and the missing array indication for the Id parameter are errors which I can correct.

All the other remarks involve the formatting that VSC is doing automatically.
The problem here might be that the removal of the .vscode folder (see #29) means there is no project standard formatting file. Which might make it very hard to have contributors follow all the (undocumented) rules.

**Comment by SimeonGerginov** [comment]
Fix those two, I will update the other things related to the formatting.

**Pull request merged**